pennylane/devices/qubit/apply_operation.py

```diff
@@ -385,7 +388,13 @@ def _evolve_state_vector_under_parametrized_evolution(
             "You can update these values by calling the ParametrizedEvolution class: EV(params, t)."
         )
 
-    state = state.flatten()
+    if is_state_batched:
+        batch_dim = state.shape[0]
+        state = qml.math.moveaxis(state.reshape((batch_dim, 2**num_wires)), 1, 0)
```

Comment:
did we have this kind of batch-dim-last logic on the legacy device, or is this all-new code? I'm surprised to see state-evolution that works this way

Reply:
We had this before already, depending on the size of the gate and the device we would either evolve the state directly, or generate the unitary of the sub-system and apply it accordingly 🙂

Reply:
sorry, lemme clarify what I mean. I checked out this branch, copied `test_with_batched_state` locally, then ran this:
```python
dev = qml.device("default.qubit.jax", wires=num_state_wires)

@qml.qnode(dev)
def c():
    qml.StatePrep(initial_state.reshape(3, -1), wires=range(num_state_wires))
    qml.apply(op)
    return qml.state()

np.allclose(c(), new_state.reshape(3, -1))
```
With `num_state_wires=4`, it evolves the matrix in the default manner and everything passes. with `num_state_wires=2`, it fails, saying:
> TypeError: dot_general requires contracting dimensions to have the same shape, got (4,) and (12,).

in other words, we never supported batching with the custom state-evolution kernel on the old device, so this is a new feature.

I don't know how the jax `odeint` works, but it clearly expects the batch dimension to be at the end instead of the start, so this is fine. I guess my comment was more of a remark than a question - I'd say our testing is sufficient to prove it works 😄

Reply:
Oh yes! We _thought_ this was possible before but it apparently wasnt. In a way, this is a new feature yeah 😄

Reply:
Yup, this is new support. It was easy to add, so I did it :)